numpy/random/_generator.pyi

```diff
@@ -0,0 +1,245 @@
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+class Generator:
+    # COMPLETE
+    _bit_generator: BitGenerator
+    _poisson_lam_max: float
+    def __init__(self, bit_generator: BitGenerator) -> None: ...
+    def __repr__(self) -> str: ...
+    def __str__(self) -> str: ...
+    # Pickling support:
+    def __getstate__(self) -> Dict[str, Any]: ...
+    def __setstate__(self, state: Dict[str, Any]): ...
+    def __reduce__(self) -> Tuple[Callable[[str], BitGenerator], Tuple[str], Dict[str, Any]]: ...
+    @property
+    def bit_generator(self) -> BitGenerator: ...
+    def bytes(self, length: int) -> str: ...
+    # TODO: Needs overloading
+    def standard_cauchy(
+        self, size: Union[None, _ShapeLike] = ...
+    ) -> Union[float, ndarray[Any, dtype[float64]]]: ...
+    # TODO: Needs overloading and specific dtypes
+    def standard_exponential(
+        self,
+        size: Optional[Union[_ShapeLike]] = ...,
+        dtype: DTypeLike = ...,
+        method: Literal["zig", "inv"] = ...,
+        out=Union[None, ndarray[Any, dtype[float32]], ndarray[Any, dtype[float64]]],
+    ) -> Union[float, ndarray[Any, Any]]: ...
+    # TODO: Needs typing
+    def random(
+        self,
+        size: Optional[_ShapeLike] = ...,
+        dtype: DTypeLike = ...,
+        out: Union[None, ndarray[Any, dtype[float32]], ndarray[Any, dtype[float64]]] = ...,
+    ): ...
+    def beta(
+        self, a: _ArrayLikeFloat_co, b: _ArrayLikeFloat_co, size: Union[None, _ShapeLike] = ...
+    ): ...
+    def exponential(self, scale: _ArrayLikeFloat_co = ..., size: Union[None, _ShapeLike] = ...): ...
+    def integers(
+        self,
+        low: _ArrayLikeInt_co,
+        high: Optional[_ArrayLikeInt_co] = ...,
+        size: Union[None, _ShapeLike] = ...,
+        dtype=...,
+        endpoint: bool = ...,
+    ) -> ndarray[Any, dtype[integer]]: ...
+    # TODO: Use a TypeVar _T here to get away from Any output?  Should be int->ndarray[Any,dtype[int64]], ArrayLike[_T] -> Union[_T, ndarray[Any,Any]]
+    def choice(
+        self,
+        a: ArrayLike,
+        size: Union[None, _ShapeLike] = ...,
+        replace: bool = ...,
+        p=...,
+        axis=...,
+        shuffle: bool = ...,
+    ) -> Any: ...
+    def uniform(
+        self,
+        low: _ArrayLikeFloat_co = ...,
+        high: _ArrayLikeFloat_co = ...,
+        size: Union[None, _ShapeLike] = ...,
+    ) -> Union[float, ndarray[Any, dtype[float64]]]: ...
+    @overload
+    def standard_normal(
+        self,
+        size: None = ...,
+        dtype: DTypeLike = ...,
+        out: None = ...,
+    ) -> float: ...
+    # TODO: How to literal dtype?
+    @overload
+    def standard_normal(
+        self,
+        size: _ShapeLike = ...,
+        dtype: DTypeLike = ...,
+        out: Union[None, ndarray[Any, dtype[float32]], ndarray[Any, dtype[float64]]] = ...,
+    ) -> Union[ndarray[Any, dtype[float32]], ndarray[Any, dtype[float64]]]: ...
```

Comment:
It might be helpful to create a dedicated `_DTypeLikeFloat32` and `_DTypeLikeFloat64` unions similar to `_DTypeLikeFloat`. https://github.com/numpy/numpy/blob/a5dc2b5b917fc50575e10bbe139a0c78e43a1c1c/numpy/typing/_dtype_like.py#L168-L180

Option B is just to annotate the dtype as `_DTypeLikeFloat`. This is a bit too broad, but it's already a big improvement over just plain `DTypeLike`.

Reply:
Do these look right?

```python
_DTypeLikeFloat32 = Union[
    Type[float32],
    Type[single],
    _Float32Codes,
    _SingleCodes,
]

_DTypeLikeFloat64 = Union[
    Type[double],
    Type[float],
    Type[float64],
    _Float64Codes,
    _DoubleCodes,
]
```

Reply:
Are `single` and `float32` identical here?  Same for `double` and `float64`?

Reply:
Yes they are, `single` and `float32` for example are both defined as `floating[_32Bit]`.
So I'd suggest to just stick to `float32` and `float64`.

Reply:
I've updated these to only use the `floatXX` variants, except for `Codes`